src/Microsoft.Azure.WebJobs.Host/Singleton/SingletonListener.cs

```diff
@@ -116,7 +118,7 @@ private void OnLockTimer(object sender, ElapsedEventArgs e)
 
         internal async Task TryAcquireLock()
         {
-            _lockHandle = await _singletonManager.TryLockAsync(_lockId, null, _attribute, CancellationToken.None, retry: false);
+            _lockHandle = await _singletonManager.TryLockAsync(_lockId, _hostMachineId, null, _attribute, CancellationToken.None, retry: false);
```

Comment:
If we're relying on it being a guid, can we change _hostMachineId's type from string to a guid? 
For example, even "string that's really a guid"  is dangerous - what are the casing rules, and are dashes allowed?   Some azure things (like blob names)  are case *sensitive*, so the same Guid could be converted to different strings. 

---
In reply to: [92680067](https://github.com/Azure/azure-webjobs-sdk/pull/913#discussion_r92680067) [](ancestors = 92680067)